### PR TITLE
Allow excluding public .NET types from being exposed to Python via [PyExport(false)]

### PR DESCRIPTION
**Pull request by lostmsu**
### What does this implement/fix? Explain your changes.

Introduced [`PyExportAttribute`](https://github.com/pythonnet/pythonnet/wiki/PyExport), that can be applied to individual types or the entire assembly, and controls visibility of these types/assemblies to Python. This is useful in case .NET type has a name conflict with Python type.

The attribute in checked in `ScanAssembly`.

### Checklist

Check all those that are applicable and complete.

-   [x] Make sure to include one or more tests for your change
-   [x] If an enhancement PR, please create docs and at best an example
-   [x] Add yourself to [`AUTHORS`](../blob/master/AUTHORS.md)
-   [x] Updated the [`CHANGELOG`](../blob/master/CHANGELOG.md)

### CODE REVIEW

**Comment by lostmsu** [comment]
My sanity is on the brink. This change does not even touch any Python objects, there are not changes in C# tests at all, and yet they start segfaulting. UGHR

File: setup.py [hunk 1]
```diff
@@ -316,6 +317,7 @@ def build_extension(self, ext):
             _config = "{0}Mono".format(CONFIG)
             _solution_file = "pythonnet.sln"
             _custom_define_constants = False
+            defines.append("NET40")
```
Comment (filmor) [inline setup.py:320]: This is new, is this intentional?

**Comment by filmor** [comment]
You are unconditionally setting `NET40` now, even on Mono, maybe that's the problem?

File: setup.py (hunk 1 above)
Comment (lostmsu) [reply setup.py:320]: @filmor yes, it is. This is used in the same commit.

**Comment by lostmsu** [comment]
@filmor we don't have any other `NET40` checks in the code, just the one I've added in this change. If this check would not have worked properly, the code would not compile due to a conflict with existing `GetCustomAttribute` extension method.

**Comment by lostmsu** [comment]
Temporarily removed `NET40` check to ensure it is required.

**Comment by lostmsu** [comment]
@filmor it seems to be passing without `NET40`, but I am still sad that I have no idea why it failed with it.

**Pull request merged**